client/keys/add.go

```diff
@@ -118,8 +118,7 @@ func runAddCmd(ctx client.Context, cmd *cobra.Command, args []string, inBuf *buf
 	}
 
 	if dryRun, _ := cmd.Flags().GetBool(flags.FlagDryRun); dryRun {
-		// use in memory keybase
-		kb = keyring.NewInMemory()
+		kb = keyring.NewInMemory(ctx.Codec)
```

Comment:
let's keep the old comment

```suggestion
		// use in memory keybase
		kb = keyring.NewInMemory(ctx.Codec)
```